### ISSUE TITLE
Variable Width Bar Chart

### ISSUE THREAD

**Issue by robin-garnham** [recharts/recharts#1199]
First of all, I love this library, it's really well put together and it's made my life so much easier.

I have a bit of a different requirement for one specific chart - I want to have a bar chart but be able to vary the width of the the individual bars.

We are using the area and the height to represent seperate metrics, so being able to vary the width is essential for my use case.

I am wondering if there is a hook I can use or if I can override a rendering method? Any guidance would be greatly appreciated!

**Comment by mitchellcunha** [comment]
@robin-garnham Agreed. I have a use case where I would like each bar to represent a span of time in hours, and have the bar widths reflect that visually. I can't seem to find an example or way to do this yet.

**Comment by Simek** [comment]
Still an issue. Any idea how to archive variable width bars on time based XAxis?

**Comment by xile611** [comment]
Sorry for not handling your issue in time, please try the latest version. If the problem persists, please open a new issue according to the issue guide.

**Comment by its-kayes** [comment]
> @robin-garnham Agreed. I have a use case where I would like each bar to represent a span of time in hours, and have the bar widths reflect that visually. I can't seem to find an example or way to do this yet.
> 
> ![screen shot 2018-04-09 at 9 40 22 am](https://user-images.githubusercontent.com/12757306/38510477-1a1aab8e-3bda-11e8-8171-8ebe6df909cb.png)
> 
> It would be great if the bars in the above chart could be dynamically widened based on the time period they represent.

Can I get a code example for this chart ?

**Comment by Vurmevska** [comment]
> Sorry for not handling your issue in time, please try the latest version. If the problem persists, please open a new issue according to the issue guide.

Can you please specify how was the issue handled ? 
Is there a way to display bars with different widths based on the x-axis values with the current version ?

**Comment by ckifer** [comment]
Reopened

**Comment by gp919934** [comment]
Is this issue got resolved?How to fix this
